### ISSUE TITLE
Intersection not supported

### ISSUE THREAD

**Issue by grabbou** [brigand/babel-plugin-flow-react-proptypes#75]
With type:
```
type Props = TransitionerProps & {
  abc: boolean
}
```
I am getting error that `Did not find type annotation for xxxx`

Here's file: https://github.com/react-community/react-navigation/blob/master/src/views/Header.js#L31

**Comment by brigand** [comment]
I put in a hack to make this kinda work. It looks at the last intersection type, and if it's an object, it uses that.

I don't have much time to work on this project, but if you want to implement a better solution, I'd be happy to review it :-)

**Comment by brigand** [comment]
Published as 1.1.0.

**Comment by grabbou** [comment]
Another issue that unfortunately doesn't work is: https://github.com/facebook/react-native/blob/0.43-stable/Libraries/CustomComponents/Lists/VirtualizedList.js#L142

```
error: bundling: Error: /Users/grabbou/Repositories/react-navigation/node_modules/react-native/Libraries/CustomComponents/Lists/VirtualizedList.js: Did not find type annotation for VirtualizedList
```

I'll try to look into that later. Feel free to suggest any intermediate workaround.

**Comment by brigand** [comment]
@grabbou [disabling it for a given file](https://github.com/brigand/babel-plugin-flow-react-proptypes#suppression) should always make errors go away.

Is this plugin running on things in your node_modules? That's quite unusual and probably something you should disable.

**Comment by grabbou** [comment]
Yeah, already disabled it, but in React Native, Node modules are transpiled
by default, which causes aforementioned problem.

Anyway, just leaving it here so we have another reproduction! 👌

**Comment by davidfurlong** [comment]
Getting this issue with

```js

export type Props = {
  onChange: Function,
  className?: string,
  value?: any // TODO! we are sometimes passing a string, sometimes an object
};

export class Address extends Component {
  static defaultProps = {
    value: null,
  }

  constructor(){
    super();
    this.state = {
      loading: true,
      loadFail: false
    };
  }

  props: Props;
...
}
```

**Comment by davidfurlong** [comment]
Found that adding `props: Props` argument to my constructor resolved the issue. Now to fix the lint error 👍

**Comment by jedwards1211** [comment]
@davidfurlong try explicitly declaring polymorphic types i.e. `Component<DefaultProps, Props, void>`.  Recent versions of flow should be reporting an error when you don't declare them anyway

**Comment by davidfurlong** [comment]
@jedwards1211 thanks mate! Will try that.

**Comment by mhaas** [comment]
Should be fixed by #87
